src/mono/netcore/System.Private.CoreLib/src/System/Environment.iOS.cs

```diff
@@ -0,0 +1,115 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Runtime.InteropServices;
+
+namespace System
+{
+    partial class Environment
+    {
+        private const int NSDocumentDirectoryId = 9;
+        private const int NSLibraryDirectoryId = 5;
+
+        private static string s_document;
+        private static string s_library;
+
+        // TODO: fix for tvOS
+        // The "normal" NSDocumentDirectory is a read-only directory on tvOS
+        // and that breaks a lot of assumptions in the runtime and the BCL
+        private static string NSDocumentDirectory => s_document ??= Interop.Sys.SearchPath(NSDocumentDirectoryId);
+
+        // Various user-visible documentation, support, and configuration files
+        private static string NSLibraryDirectory => s_library ??= Interop.Sys.SearchPath(NSLibraryDirectoryId);
```

Comment:
do we really want to initialize these in the static .cctor? most apps probably won't use `Environment.GetFolderPath` but `Environment` is likely in the app startup path so we could move the initialization to `GetFolderPathiOS`

Reply:
@akoeplinger um.. it's a property 🙂

Reply:
ah lol, didn't see it was `=>` and not `=` :)